diff -rupN gtk/cheatsGTK.cpp gtk-patched/cheatsGTK.cpp
--- gtk/cheatsGTK.cpp	2010-11-17 03:05:43.000000000 -0500
+++ gtk-patched/cheatsGTK.cpp	2014-02-10 09:20:35.144498207 -0500
@@ -191,6 +191,55 @@ static void cheat_list_add_cheat(GtkWidg
 #undef NEW_DESC
 }
 
+static GtkListStore *cheat_list_populate()
+{
+    GtkListStore *store = gtk_list_store_new (5, G_TYPE_BOOLEAN, 
+            G_TYPE_INT, G_TYPE_INT, G_TYPE_INT, G_TYPE_STRING);
+
+    CHEATS_LIST cheat;
+    u32 chsize = cheats->getSize();
+    for(u32 ii = 0; ii < chsize; ii++){
+        GtkTreeIter iter;
+        cheats->get(&cheat, ii);
+        gtk_list_store_append(store, &iter);
+        gtk_list_store_set(store, &iter,
+                COLUMN_ENABLED, cheat.enabled,
+                COLUMN_SIZE, cheat.size+1,
+                COLUMN_HI, cheat.code[0][0],
+                COLUMN_LO, cheat.code[0][1],
+                COLUMN_DESC, cheat.description,
+                -1);
+    }
+    return store;
+}
+
+static void cheat_list_refresh_list(GtkTreeView *tree)
+{
+    GtkListStore *store = cheat_list_populate();
+    gtk_tree_view_set_model(tree, GTK_TREE_MODEL(store));
+}
+
+static void cheat_list_add_AR_cheat(GtkWidget * widget, gpointer entry)
+{
+    /** Add an Action Replay cheat code to the cheats list */
+    if (!cheats->add_AR(const_cast<char *>(gtk_entry_get_text(GTK_ENTRY(entry))), "AR Cheat Code", TRUE))
+      {
+
+        GtkWidget *dialog = gtk_message_dialog_new (GTK_WINDOW(win),
+                                          GTK_DIALOG_DESTROY_WITH_PARENT,
+                                          GTK_MESSAGE_ERROR,
+                                          GTK_BUTTONS_CLOSE,
+                                          "Please enter a valid Action Replay code");
+        gtk_dialog_run (GTK_DIALOG (dialog));
+        gtk_widget_destroy (dialog);
+      }
+
+    GtkTreeView *tree = GTK_TREE_VIEW(g_object_get_data(G_OBJECT(entry), "treeview"));
+    cheat_list_refresh_list(tree);
+
+    gtk_widget_show_all(win);
+}
+
 static GtkTreeModel * create_numbers_model (void)
 {
 #define N_NUMBERS 4
@@ -279,28 +328,6 @@ static void cheatListEnd()
         Launch();
 }
 
-static GtkListStore *cheat_list_populate()
-{
-    GtkListStore *store = gtk_list_store_new (5, G_TYPE_BOOLEAN, 
-            G_TYPE_INT, G_TYPE_INT, G_TYPE_INT, G_TYPE_STRING);
-
-    CHEATS_LIST cheat;
-    u32 chsize = cheats->getSize();
-    for(u32 ii = 0; ii < chsize; ii++){
-        GtkTreeIter iter;
-        cheats->get(&cheat, ii);
-        gtk_list_store_append(store, &iter);
-        gtk_list_store_set(store, &iter,
-                COLUMN_ENABLED, cheat.enabled,
-                COLUMN_SIZE, cheat.size+1,
-                COLUMN_HI, cheat.code[0][0],
-                COLUMN_LO, cheat.code[0][1],
-                COLUMN_DESC, cheat.description,
-                -1);
-    }
-    return store;
-}
-
 static GtkWidget *cheat_list_create_ui()
 {
     GtkListStore *store = cheat_list_populate();
@@ -321,6 +348,16 @@ static GtkWidget *cheat_list_create_ui()
     g_signal_connect (button, "clicked", G_CALLBACK (cheat_list_remove_cheat), tree);
     gtk_container_add(GTK_CONTAINER(hbbox),button);
 
+    GtkWidget *entry;
+    entry = gtk_entry_new();
+    gtk_entry_set_text(GTK_ENTRY(entry), "Put in an Action Replay code");
+    gtk_container_add(GTK_CONTAINER(hbbox), entry);
+
+    button = gtk_button_new_with_label("Add AR cheat");
+    g_object_set_data(G_OBJECT(entry), "treeview", tree);
+    g_signal_connect (button, "clicked", G_CALLBACK (cheat_list_add_AR_cheat), entry);
+    gtk_container_add(GTK_CONTAINER(hbbox),button);
+
     cheat_list_add_columns(GTK_TREE_VIEW(tree), store);
     
     /* Setup the selection handler */
