diff -ur src.orig/gtk/glx_3Demu.cpp src/gtk/glx_3Demu.cpp
--- src.orig/gtk/glx_3Demu.cpp	2013-11-27 19:36:53.712615822 -0500
+++ src/gtk/glx_3Demu.cpp	2014-09-16 12:48:41.424818929 -0400
@@ -31,7 +31,7 @@
 static GLXContext ctx;
 static GLXPbuffer pbuf;
 
-void deinit_glx_3Demu(void)
+bool deinit_glx_3Demu(void)
 {
     Display *dpy = glXGetCurrentDisplay();
 
diff -ur src.orig/gtk/glx_3Demu.h src/gtk/glx_3Demu.h
--- src.orig/gtk/glx_3Demu.h	2013-11-27 19:36:53.716615886 -0500
+++ src/gtk/glx_3Demu.h	2014-09-16 12:49:25.040818408 -0400
@@ -20,5 +20,5 @@
 
 #ifdef HAVE_GL_GLX
 int  init_glx_3Demu(void);
-void deinit_glx_3Demu(void);
+bool deinit_glx_3Demu(void);
 #endif
