--- shairport-sync-4.2/Makefile.am.orig	2023-09-11 13:34:59.682314707 -0400
+++ shairport-sync-4.2/Makefile.am	2023-09-11 13:36:12.737176491 -0400
@@ -277,7 +277,7 @@
 # will be stored in a scripts folder in the _build_ folder
 # which will be the source folder if you're not using a separate build folder
 
-$(INSTALL_SYSTEMV_TARGET): scripts/shairport-sync $(INSTALL_USER_TARGET)
+$(INSTALL_SYSTEMV_TARGET): scripts/shairport-sync
 	install -d $(DESTDIR)$(sysconfdir)/init.d
 	[ -e $(DESTDIR)$(sysconfdir)/init.d/shairport-sync ] || install -m 0755 scripts/shairport-sync $(DESTDIR)$(sysconfdir)/init.d
 
@@ -297,7 +297,7 @@
 # will be stored in a scripts folder in the _build_ folder
 # which will be the source folder if you're not using a separate build folder
 
-$(INSTALL_SYSTEMD_TARGET): scripts/$(SYSTEMD_SERVICE) $(INSTALL_USER_TARGET)
+$(INSTALL_SYSTEMD_TARGET): scripts/$(SYSTEMD_SERVICE)
 	install -d $(DESTDIR)$(systemdsystemunitdir)
 	[ -e $(DESTDIR)$(systemdsystemunitdir)/shairport-sync.service ] || install -m 0644 scripts/$(SYSTEMD_SERVICE) $(DESTDIR)$(systemdsystemunitdir)/shairport-sync.service
 
