diff -ur noto-emoji-old/Makefile noto-emoji-aac7ccaa4d1dea4543453b96f7d6fc47066a57ff/Makefile
--- noto-emoji-old/Makefile	2020-12-06 16:09:05.480529878 -0500
+++ noto-emoji-aac7ccaa4d1dea4543453b96f7d6fc47066a57ff/Makefile	2020-12-06 16:15:00.105726457 -0500
@@ -188,7 +188,7 @@
 	@($(PNGQUANT) $(PNGQUANTFLAGS) -o "$@" "$<"; case "$$?" in "98"|"99") echo "reuse $<";cp $< $@;; *) exit "$$?";; esac)
 
 $(COMPRESSED_DIR)/%.png: $(QUANTIZED_DIR)/%.png | check_tools $(COMPRESSED_DIR)
-	@$(ZOPFLIPNG) -y "$<" "$@" 1> /dev/null 2>&1
+	@$(ZOPFLIPNG) -y "$<" "$@" 2>&1
 
 # Make 3.81 can endless loop here if the target is missing but no
 # prerequisite is updated and make has been invoked with -j, e.g.:
