diff -ur kanboard-1.0.18/Makefile kanboard-1.0.18-new/Makefile
--- kanboard-1.0.18/Makefile	2015-08-30 12:30:30.000000000 -0400
+++ kanboard-1.0.18-new/Makefile	2015-09-16 23:48:06.302428952 -0400
@@ -13,19 +13,19 @@
 vendor.css:
 	@ cat ${CSS_VENDOR} > vendor.css
 
-app.css:
+app.css: vendor.css
 	@ rm -f assets/css/app.css
-	@ cat ${CSS_APP} > tmp.css
-	@ yuicompressor --charset utf-8 --type css -o tmp.css tmp.css
-	@ cat vendor.css tmp.css >> assets/css/app.css
-	@ rm -f tmp.css
+	@ cat ${CSS_APP} > apptmp.css
+	@ yuicompressor --charset utf-8 --type css -o apptmp.css apptmp.css
+	@ cat vendor.css apptmp.css >> assets/css/app.css
+	@ rm -f apptmp.css
 
-print.css:
+print.css: vendor.css
 	@ rm -f assets/css/print.css
-	@ cat ${CSS_PRINT} > tmp.css
-	@ yuicompressor --charset utf-8 --type css -o tmp.css tmp.css
-	@ cat vendor.css tmp.css >> assets/css/print.css
-	@ rm -f tmp.css
+	@ cat ${CSS_PRINT} > printtmp.css
+	@ yuicompressor --charset utf-8 --type css -o printtmp.css printtmp.css
+	@ cat vendor.css printtmp.css >> assets/css/print.css
+	@ rm -f printtmp.css
 
 css: vendor.css app.css print.css
 	@ rm -f vendor.css
@@ -34,14 +34,14 @@
 	@ cat ${JS_VENDOR} > vendor.js
 	@ cat ${JS_LANG} >> vendor.js
 
-app.js:
+app.js: vendor.js
 	@ rm -f assets/js/app.js
-	@ echo "(function() { 'use strict';" > tmp.js
-	@ cat ${JS_APP} >> tmp.js
-	@ echo "})();" >> tmp.js
-	@ yuicompressor --charset utf-8 --type js -o tmp.js tmp.js
-	@ cat vendor.js tmp.js >> assets/js/app.js
-	@ rm -f tmp.js
+	@ echo "(function() { 'use strict';" > apptmp.js
+	@ cat ${JS_APP} >> apptmp.js
+	@ echo "})();" >> apptmp.js
+	@ yuicompressor --charset utf-8 --type js -o apptmp.js apptmp.js
+	@ cat vendor.js apptmp.js >> assets/js/app.js
+	@ rm -f apptmp.js
 
 js: vendor.js app.js
 	@ rm -f vendor.js
