--- a/install.sh	2012-07-03 12:54:47.000000000 -0400
+++ b/install.sh	2014-07-01 12:36:00.679412366 -0400
@@ -10,6 +10,7 @@
 	PREFIX=$HOME
 fi
 installverify=yes
+ebuild=no
 BC='beyondcompare'
 BCICON='bcompare.png'
 BC_CONTEXT_FULL='bcomparefull32.png'
@@ -30,7 +31,9 @@
 						  /usr, /usr/local, /opt/beyondcompare, and \$HOME.
 						  Files will be installed in '\$PREFIX/bin' and
 						  '\$PREFIX/lib/beyondcompare'.
-  --help				prints out this message
+  --ebuild			do not install anything to /usr/share (ebuild compatibility)
+  --libdirname[=DIRNAME]	change the library directory name
+  --help			prints out this message
 EOF
 	exit $1
 }
@@ -50,6 +53,12 @@
 			PREFIX=$optarg
 			installverify=no
 			;;
+		--ebuild)
+			ebuild=yes
+			;;
+		--libdirname=*)
+			BC=$optarg
+			;;
 		--help)
 			usage 1 1>&2
 			;;
@@ -110,14 +119,20 @@
 ##
 # insert these variables in the other scripts
 ##
-sed -i "s|^\(BC_LIB=\).*|\1$BC_LIB|" bcompare.sh
-sed -i "s|^\(BC_PACKAGE_TYPE=\).*|\1$BC_PACKAGE_TYPE|" bcompare.sh
-sed -i "s|^\(BC_LIB=\).*|\1$BC_LIB|" kde_context_menu.sh
-sed -i "s|^\(BC_BIN=\).*|\1$BC_BIN|" kde_context_menu.sh
-sed -i "s|^\(BC_LIB=\).*|\1$BC_LIB|" uninstall.sh
-sed -i "s|^\(BC_BIN=\).*|\1$BC_BIN|" uninstall.sh
-sed -i "s|^\(PREFIX=\).*|\1$PREFIX|" uninstall.sh
-
+if [ ${ebuild} = no ]; then
+	sed -i "s|^\(BC_LIB=\).*|\1$BC_LIB|" bcompare.sh
+	sed -i "s|^\(BC_PACKAGE_TYPE=\).*|\1$BC_PACKAGE_TYPE|" bcompare.sh
+	sed -i "s|^\(BC_LIB=\).*|\1$BC_LIB|" kde_context_menu.sh
+	sed -i "s|^\(BC_BIN=\).*|\1$BC_BIN|" kde_context_menu.sh
+	sed -i "s|^\(BC_LIB=\).*|\1$BC_LIB|" uninstall.sh
+	sed -i "s|^\(BC_BIN=\).*|\1$BC_BIN|" uninstall.sh
+	sed -i "s|^\(PREFIX=\).*|\1$PREFIX|" uninstall.sh
+else
+	sed -i "s|^\(BC_LIB=\).*|\1${BC_LIB#${D}}|" bcompare.sh
+	sed -i "s|^\(BC_PACKAGE_TYPE=\).*|\1$BC_PACKAGE_TYPE|" bcompare.sh
+	sed -i "s|^\(BC_LIB=\).*|\1${BC_LIB#${D}}|" kde_context_menu.sh
+	sed -i "s|^\(BC_BIN=\).*|\1${BC_LIB#${D}}|" kde_context_menu.sh
+fi
 
 ##
 # Copy the files
@@ -171,6 +186,12 @@
 if [ "$smaller" = "$testver" ]; then return 1; else return 0; fi
 }
 
+##
+# Stop here if we are in ebuild mode. Do not install anything in /usr/share 
+##
+if [ ${ebuild} = yes ]; then
+	exit 0;
+fi
 
 ##
 # Create desktop entry and place Context Menu extensions, if root
